gapic-generator-java/DEVELOPMENT.md

```diff
@@ -1,210 +1,145 @@
-    ```
-    or
-    ```sh
-    mvn fmt:check
-    ```
-
--   Format files.
-
-    ```sh
-    bazel run //:google_java_format
-    ```
-    or
-    ```sh
-    mvn fmt:format
-    ```
-
 ## Test Running
 
--   Run all unit and integration tests.
+- Run all unit and integration tests.
 
     ```sh
-    bazel test //...
+    mvn install          # unit tests, maven test wouldn't work in root folder because gapic-generator-java is dependant on test jars of gax-java
+    bazel test //...  # integration tests
     ```
+- For running unit tests in `gapic-generator-java` submodule, first build all modules with `mvn install -DskipTests`, then `cd` into `gapic-generator-java` submodule for the following commands:
+    - Run a single or multiple unit tests:
 
--   Run all unit tests.
+      ```sh
+      mvn test -Dtest=JavaCodeGeneratorTest
+  
+      mvn test "-Dtest=Basic*, !%regex[.*.Unstable.*], !%regex[.*.MyTest.class#one.*|two.*], %regex[#fast.*|slow.*]"
+      ```
 
-    ```sh
-    bazel test //:units
-    ```
-    or
-    ```sh
-    mvn test
-    ```
+    - Update all unit test golden files:
 
--   Run a single unit test like `JavaCodeGeneratorTest.java`:
+      ```sh
+      mvn test -DupdateUnitGoldens
+      ```
 
-    ```sh
-    bazel test //:unit_com_google_api_generator_engine_JavaCodeGeneratorTest
-    ```
-    or
-    ```sh
-    mvn test -Dtest=JavaCodeGeneratorTest
+    - Update a single unit test golden file, for example `JavaCodeGeneratorTest.java`:
 
-    mvn test "-Dtest=Basic*, !%regex[.*.Unstable.*], !%regex[.*.MyTest.class#one.*|two.*], %regex[#fast.*|slow.*]"
-    ```
-￼
--   Update all unit test golden files:
-
-    ```sh
-    mvn test -DupdateUnitGoldens
-    ```
-
--   Update a single unit test golden file, for example `JavaCodeGeneratorTest.java`:
-
-    ```sh
-    bazel run //:update_com_google_api_generator_engine_JavaCodeGeneratorTest
-    ```
-    or
-    ```sh
-    mvn test -DupdateUnitGoldens -Dtest=JavaCodeGeneratorTest
-    ```
+      ```sh
+      mvn test -DupdateUnitGoldens -Dtest=JavaCodeGeneratorTest
+      ```
 
--   Run a single integration test for API like `Redis`, it generates Java source
-    code using the Java microgenerator and compares them with the goldens files
-    in `test/integration/goldens/redis`.
+- Run a single integration test for API like `Redis`, it generates Java source
+  code using the Java microgenerator and compares them with the goldens files
+  in `test/integration/goldens/redis`.
 
     ```sh
     bazel test //test/integration:redis
     ```
 
--   Update integration test golden files, for example `Redis`. This clobbers all the
-    files in `test/integration/goldens/redis`.
+- Update integration test golden files, for example `Redis`. This clobbers all the
+  files in `test/integration/goldens/redis`.
 
     ```sh
     bazel run //test/integration:update_redis
     ```
 
-## Running the Plugin
+## Running the Plugin under googleapis with local gapic-generator-java
```

Comment:
I think we could move this whole section to root folder also, as we can use this way to test local gax changes as well.

Reply:
I prefer keeping DEVELOPMENT.md specific to module. To address test local gax change, I created gax-java/DEVELOPMENT.md that points to gapic-generator-java/DEVELOPMENT.md for local testing.